### PR TITLE
Transaction failure in graphql api

### PR DESCRIPTION
**Pull request by imeckler**
This PR exposes transaction failure reason from graphQL (as mentioned here: https://github.com/MinaProtocol/mina/issues/8151)

### CODE REVIEW

File: src/lib/mina_graphql/mina_graphql.ml [hunk 1]
```diff
@@ -1231,29 +1252,38 @@ module Types = struct
...
           ~deprecated:(Deprecated (Some "use receiver field instead"))
           ~args:Arg.[]
           ~resolve:(fun {ctx= coda; _} cmd ->
             AccountObj.get_best_ledger_account coda
             @@ Signed_command.receiver ~next_available_token:Token_id.invalid
-                 cmd.With_hash.data ) ]
+                 cmd.With_hash.data )
+      ; field "failureReason" ~typ:string ~args:[]
+          ~doc:
+            "null is no failure or status unknown, reason for failure \
```
Comment (psteckler) [inline src/lib/mina_graphql/mina_graphql.ml:1280]: there's a missing word or something here

File: src/lib/mina_graphql/mina_graphql.ml [hunk 2]
```diff
@@ -1151,47 +1151,68 @@ module Types = struct
...
+        | Unknown
+    end
+
+    module With_status = struct
+      type 'a t = {data: 'a; status: Status.t}
+
+      let map t ~f = {t with data= f t.data}
+    end
+
+    let field' ?doc ?deprecated lab ~typ ~args ~resolve =
```
Comment (psteckler) [inline src/lib/mina_graphql/mina_graphql.ml:1171]: would this be better-named as something like `field_with_status_data`

Comment (bkase) [reply src/lib/mina_graphql/mina_graphql.ml:1171]: I like `field'` actually since it's used so often

**Review by bkase — APPROVED** [review]

**Comment by bkase** [comment]
!approved-for-mainnet

**Comment by bkase** [comment]
Closes #8151 and #5960

**Comment by psteckler** [comment]
> I like field' actually since it's used so often

But it doesn't suggest its purpose.

**Comment by imeckler** [comment]
> > I like field' actually since it's used so often
> 
> But the name doesn't suggest its purpose.

That makes sense to me -- made this change

**Pull request merged**